istioctl/cmd/root.go

```diff
@@ -74,6 +75,18 @@ func defaultLogOptions() *log.Options {
 	return o
 }
 
+// ConfigAndEnvProcessing uses spf13/viper for overriding CLI parameters
+func ConfigAndEnvProcessing() error {
+	// Allow users to override some variables through $HOME/.istioctl/config.yaml
+	// and environment variables.
+	viper.SetEnvPrefix("ISTIOCTL")
+	viper.AutomaticEnv()
+	viper.SetConfigName("config") // name of config file (without extension)
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath("$HOME/.istioctl")
```

Comment:
I think this may break on windows? Maybe we should use https://github.com/mitchellh/go-homedir.

Reply:
Viper turns out to be very smart about this, see https://github.com/spf13/viper/blob/master/util.go#L138